### ISSUE TITLE
open5gs stopped & adb: device 'UWEUW4XG8XCA8PWS' not found

### ISSUE THREAD

**Issue by lpj12121** [asset-group/5ghoul-5g-nr-attacks#19]
I installed open5gs, and successfully started usrp, followed the steps[https://github.com/asset-group/5ghoul-5g-nr-attacks#2--quick-start-docker-container](url) to start the rogue base station, but encountered a problem：

[SHMDriver] SHM:/tmp/wshm, Channel:1, Mode:1
[SHMDriver] SHM:/tmp/wshm, Channel:2, Mode:1
[SHMDriver] SHM:/tmp/wshm, Channel:3, Mode:1
[SHMDriver] SHM:/tmp/wshm, Channel:4, Mode:1
[SHMDriver] SHM:/tmp/wshm, Channel:5, Mode:1
[SHMDriver] SHM:/tmp/wshm, Channel:6, Mode:1
[Open5GS] Adding IMSI 001010000000001 with K=00112233445566778899AABBCCDDEEFF, OPC=00112233445566778899AABBCCDDEEFF, APN=default
[Open5GS] Adding IMSI 001010100011321 with K=12345678901234567890123456789012, OP=12345678901234561234567890123456, APN=default
* daemon started successfully
**_adb: device 'UWEUW4XG8XCA8PWS' not found_**
[Open5GS] Adding IMSI 001010000064950 with K=3ac9ec861c3d5209ddb00d88b8b2c933, OPC=ad3d5e6e6df84bf3fd799b39c70e7c74, APN=default
[Open5GS] Adding IMSI 001010000064951 with K=3ac9ec861c3d5209ddb00d88b8b2c933, OPC=ad3d5e6e6df84bf3fd799b39c70e7c74, APN=default

...
...

**_adb: device 'UWEUW4XG8XCA8PWS' not found_**
[Open5GS] Adding IMSI 901700000039908 with K=E5EB44B93317E84EBA206EF962A29D8C, OPC=4242F3B4D58A5DA39336E1F8CB643B2A, APN=internet
[Open5GS] Subscribers registered to core network: 14
./3rd-party/hostapd/idemptables -A INPUT -i ogstun -j ACCEPT
./3rd-party/hostapd/idemptables -A FORWARD ! -i ogstun -o ogstun -j ACCEPT
./3rd-party/hostapd/idemptables -A FORWARD -i ogstun ! -o ogstun -j ACCEPT
./3rd-party/hostapd/idemptables -t nat -A POSTROUTING -s 45.45.0.0/16 ! -o ogstun -j MASQUERADE
[GlobalTimeout] Not enabled in config. file
[AnomalyReport] Added Logging Sink: PacketLogger
[AnomalyReport] Added Logging Sink: SvcReportSender
[USBHubControl] Disabled in config. file
[ReportSender] Credentials file not found: modules/reportsender/credentials.json
[ReportSender] Ready
**_[!] Open5GS stopped
[!] Base-Station process stopped_**
[Optimizer] Optimization disabled. Using default population:
--------------------------------------------------------
[Optimizer] Iter=1  Params=[0.2,0.2,0.2,0.2,0.2,0.2,...,0.2]
[Optimizer] Fitness=1e+06  Adj. Fitness=-1e+06
--------------------------------------------------------
[Optimizer] Initialized with X Size=293, Population Size=5
[Main] Fuzzing not enabled! Running only target reconnection
[PacketHandler] Added "proto:nas-5gs", Dir:0, Realtime:0, TID:1297
[PacketHandler] Added "proto:nas-5gs", Dir:1, Realtime:0, TID:1298
[PacketHandler] Added "proto:pdcp-nr-framed", Dir:0, Realtime:1, TID:1299
[PacketHandler] Added "proto:pdcp-nr-framed", Dir:1, Realtime:1, TID:1300
[PacketHandler] Added "proto:mac-nr-framed", Dir:0, Realtime:1, TID:1301
[PacketHandler] Added "proto:mac-nr-framed", Dir:0, Realtime:1, TID:1303
[PacketHandler] Added "proto:mac-nr-framed", Dir:1, Realtime:0, TID:1309
_**adb: device 'UWEUW4XG8XCA8PWS' not found
[!] Open5GS stopped
[!] Open5GS stopped
[!] Open5GS stopped**_
**_adb: device 'UWEUW4XG8XCA8PWS' not found
[!] Open5GS stopped
[!] Open5GS stopped
[!] Open5GS stopped_**

Why does this happen? My system version is ubuntu22.04, but my colleagues using 18.04 also encountered these problems.

**Comment by lpj12121** [comment]
Sorry, there was something wrong when I edited the text, which caused some text fonts to be too large. Please forgive me.

**Comment by alphafox02** [comment]
Are you using a B210? 

Change true to false in 5ghoul/configs/5gnr_gnb_config.json to stop looking for the adb device that you're likely not using.
        "Monitor": {
            "Enable": false,

**Comment by lpj12121** [comment]
> Are you using a B210?
> 
> Change true to false in 5ghoul/configs/5gnr_gnb_config.json to stop looking for the adb device that you're likely not using. "Monitor": { "Enable": false,

Yeah, I am using usrp b210. Thank you very much, but I encountered new problems according to your configuration. The following is the error message:

> [Open5GS] Subscribers registered to core network: 14
./3rd-party/hostapd/idemptables -A INPUT -i ogstun -j ACCEPT
./3rd-party/hostapd/idemptables -A FORWARD ! -i ogstun -o ogstun -j ACCEPT
./3rd-party/hostapd/idemptables -A FORWARD -i ogstun ! -o ogstun -j ACCEPT
./3rd-party/hostapd/idemptables -t nat -A POSTROUTING -s 45.45.0.0/16 ! -o ogstun -j MASQUERADE
[GlobalTimeout] Not enabled in config. file
[AnomalyReport] Added Logging Sink: PacketLogger
[AnomalyReport] Added Logging Sink: SvcReportSender
[USBHubControl] Disabled in config. file
[ReportSender] Credentials file not found: modules/reportsender/credentials.json
[ReportSender] Ready
[Optimizer] Optimization disabled. Using default population:
--------------------------------------------------------

> [Optimizer] Iter=1  Params=[0.2,0.2,0.2,0.2,0.2,0.2,...,0.2]

> [Optimizer] Fitness=1e+06  Adj. Fitness=-1e+06

--------------------------------------------------------

> [Optimizer] Initialized with X Size=293, Population Size=5

> [Main] Fuzzing not enabled! Running only target reconnection
[PacketHandler] Added "proto:nas-5gs", Dir:0, Realtime:0, TID:904
[PacketHandler] Added "proto:nas-5gs", Dir:1, Realtime:0, TID:905
[PacketHandler] Added "proto:pdcp-nr-framed", Dir:0, Realtime:1, TID:906
[PacketHandler] Added "proto:pdcp-nr-framed", Dir:1, Realtime:1, TID:907
[!] Open5GS stopped
[!] Base-Station process stopped
[PacketHandler] Added "proto:mac-nr-framed", Dir:0, Realtime:1, TID:908
[PacketHandler] Added "proto:mac-nr-framed", Dir:0, Realtime:1, TID:910
[PacketHandler] Added "proto:mac-nr-framed", Dir:1, Realtime:0, TID:913
[!] Open5GS stopped
[!] Open5GS stopped
[!] Open5GS stopped
..
..
..
[!] Open5GS stopped
[!] Open5GS stopped
[ModemManager] StartModemConnection: Modem not initialized

Have you encountered these problems?

**Comment by Matheus-Garbelini** [comment]
Hi @lpj12121,

You can also disable the parameter **UEModemManager->Enable** (change from `true` to `false`) in global_config.json:
https://github.com/asset-group/5ghoul-5g-nr-attacks/blob/3e61d51b0eea21c1743d027b0a8965b925881703/configs/global_config.json#L319-L320

This will stop this error messages from ModemManager.

**Comment by shuimoshusheng** [comment]
@lpj12121 
If you are connected to your phone, you can use the adb devices command to view the device ID of your phone and replace this value with your phone's device ID. If not, set UEModermManager enable to false in the configuration file. Additionally, USRP B210 needs to be used before use` usrp_find_device` command checks if it is mounted on USB 3.0, so that the base station will not stop. If you solve it, remember to give a good review.😄
